app/views/shared/forms/_savings_and_investments_form.html.erb

```diff
@@ -1,26 +1,34 @@
 <%= render partial: 'accounts_summary', locals: { bank_accounts: bank_accounts } if bank_accounts.any? %>
 
-<%= form_with(model: @form,
-              url: form_path,
-              method: :patch,
-              local: true) do |form| %>
+<%= form_with(model: @form, url: form_path, method: :patch, local: true) do |form| %>
 
-  <%= govuk_form_group do %>
-    <div class="govuk-checkboxes" data-module="checkboxes">
-      <%= render(
-            partial: 'shared/forms/revealing_checkbox/attribute',
-            collection: attributes,
-            locals: {
-              model: @form,
-              form: form
-            }
-          ) %>
-    </div>
+  <%= govuk_form_group show_error_if: (@form.errors.present? && !@form.any_checkbox_checked?) do %>
```

Comment:
As the form method `any_checkbox_checked_or_draft` adds a base error, the code shouldn't have to check for that condition again here. 

That is - if `@form.any_checkbox_checked?` is false, there should already be an error in `@form.errors`

This should work:
```ruby
govuk_form_group show_error_if: (@form.errors.present?)
```